rust/arrow/src/compute/kernels/cast.rs

```diff
@@ -376,6 +378,27 @@ pub fn cast(array: &ArrayRef, to_type: &DataType) -> Result<ArrayRef> {
             Int64 => cast_string_to_numeric::<Int64Type>(array),
             Float32 => cast_string_to_numeric::<Float32Type>(array),
             Float64 => cast_string_to_numeric::<Float64Type>(array),
+            Date32(DateUnit::Day) => {
+                use chrono::{NaiveDate, NaiveTime};
+                let zero_time = NaiveTime::from_hms(0, 0, 0);
```

Comment:
We don't yet have convention around timezone, should we be using `Utc` instead of `NaiveTime` here? Or does it not make a difference because `Date32` only considers days since epoch?

Reply:
`.and_time(zero_time)` is only necessary because the `timestamp()` method is only available on `NaiveDateTime`; time zone shouldn't mater for dates.